catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/event/SubscriptionManager.java

```diff
@@ -0,0 +1,345 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.spatial.ogc.csw.catalog.endpoint.event;
+
+import java.io.IOException;
+import java.util.Dictionary;
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.Map;
+import java.util.UUID;
+
+import org.apache.commons.lang.StringUtils;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceReference;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.event.InvalidSubscriptionException;
+import ddf.catalog.event.Subscription;
+
+/**
+ * SubscriptionManager manages registeredSubscriptions for add, delete, and persistence.
+ */
+public class SubscriptionManager {
+    /**
+     * The log4j BASE_LOGGER for this class
+     */
+    private static final Logger LOGGER = LoggerFactory.getLogger(SubscriptionManager.class);
+
+    /**
+     * The ManagedServiceFactory PID for the subscription's callback web service adapter,
+     * used to dynamically instantiate the web service's adapter
+     */
+    private static final String FACTORY_PID = "CSW_Subscription";
+
+    private static Map<String, ServiceRegistration<Subscription>> registeredSubscriptions =
+            new HashMap<>();
+
+    private static Map<String, Subscription> subscriptions = new HashMap<>();
+
+    protected SubscriptionManager() {
+        // SubscriptionManager should be used statically
+        throw new UnsupportedOperationException();
+    }
+
+    public static boolean hasSubscription(String subscriptionId) {
```

Comment:
Instead of making everything static, can't we just create a instance in blueprint and inject it? That would make it possible to mock this guy out in our unit tests and keep the `CswEndpointTest` simpler.